File: PYTHON/ArrayAndStrings/Easy/ValidPalindrome.py
def validPalindrome(str):
    # Remove non-alphanumeric characters and convert to lowercase
    cleaned_str = ''.join(char.lower() for char in str if char.isalnum())
    # Check if the cleaned string is equal to its reverse
    return cleaned_str == cleaned_str[::-1]

# Example usage
str1 = "Was it a car or a cat I saw?"
str2 = "Hello, World!"
if __name__ == "__main__":
    print("Is the string a valid palindrome?", validPalindrome(str1))  # Output: True
    print("Is the string a valid palindrome?", validPalindrome(str2))  # Output: False

File: JAVASCRIPT/ArrayAndStrings/Easy/ValidPalindrome.js
// VALID PALINDROME
function validPalindrome(str){
    str = str.toLowerCase().replace(/[^a-z0-9]/g, '');
    let left = 0;
    let right = str.length - 1;

    while (left < right) {
        if (str[left] !== str[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

str = "Was it a car or a cat I saw?";
console.log(validPalindrome(str)); // Output: true
// Explanation: The string is a palindrome when ignoring spaces and punctuation.    
// Example with a non-palindrome
str2 = "Hello, World!";
console.log(validPalindrome(str2)); // Output: false
// Explanation: The string is not a palindrome, even when ignoring spaces and punctuation.